airflow/providers/openai/operators/openai.py

```diff
@@ -0,0 +1,75 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from __future__ import annotations
+
+from functools import cached_property
+from typing import TYPE_CHECKING, Any, Sequence
+
+from airflow.models import BaseOperator
+from airflow.providers.openai.hooks.openai import OpenAIHook
+
+if TYPE_CHECKING:
+    from airflow.utils.context import Context
+
+
+class OpenAIEmbeddingOperator(BaseOperator):
+    """
+    Operator that accepts input text to generate OpenAI embeddings using the specified model.
+
+    .. seealso::
+        For more information on how to use this operator, take a look at the guide:
+        :ref:`howto/operator:OpenAIEmbeddingOperator`
+
+    :param conn_id: The OpenAI connection.
+    :param input_text: The text to generate OpenAI embeddings on. Either input_text or input_callable
+        should be provided.
+    :param input_callable: The callable that provides the input text to generate OpenAI embeddings.
+        Either input_text or input_callable should be provided.
+    :param input_callable_args: The list of arguments to be passed to ``input_callable``
```

Comment:
we should remove the param which are not in init

Reply:
Oops, missed it. Removed now.